vehicle_gateway_python_helpers/vehicle_gateway_python_helpers/helpers.py

```
@@ -0,0 +1,177 @@
+        additional_env=additional_env,
+        output='screen')
+
+    return run_px4
+
+
+class WorldPoseFromSdfFrame(Substitution):
+    """Substitution that retrieves a frame from SDF."""
+
+    def __init__(self,
+                 frame_name: SomeSubstitutionsType,
+                 world_name: SomeSubstitutionsType,
+                 model_pose: SomeSubstitutionsType,
+                 coor_name: str) -> None:
+        super().__init__()
+
+        from launch.utilities import normalize_to_list_of_substitutions  # import here to avoid loop
+        self.__frame_name = normalize_to_list_of_substitutions(frame_name)
+        self.__world_name = normalize_to_list_of_substitutions(world_name)
+        self.__model_pose = normalize_to_list_of_substitutions(model_pose)
+        self.__x = '0.0'
+        self.__y = '0.0'
+        self.__z = '0.3'
+        self.__roll = '0.0'
+        self.__pitch = '0.0'
+        self.__yaw = '0.0'
+        self.__coord_name = coor_name
+
+    @property
+    def model_pose(self) -> List[Substitution]:
+        """Getter for model pose."""
+        return self.__model_pose
+
+    @property
+    def frame_name(self) -> List[Substitution]:
+        """Getter for frame name."""
+        return self.__frame_name
+
+    @property
+    def world_name(self) -> List[Substitution]:
+        """Getter for world name."""
+        return self.__world_name
+
+    def parseCoords(self, strCoords: str, key: str, strSplit: str):
+        x, y, z, roll, pitch, yaw = strCoords.split(strSplit)
+        if (self.__coord_name == 'x'):
+            return str(x)
+        if (self.__coord_name == 'y'):
+            return str(y)
+        if (self.__coord_name == 'z'):
+            return str(z)
+        if (self.__coord_name == 'roll'):
+            return str(roll)
+        if (self.__coord_name == 'pitch'):
+            return str(pitch)
+        if (self.__coord_name == 'yaw'):
+            return str(yaw)
+        return '0.0'
```

I think you should raise an error if none of the if statements match. This seems like a sneaky bug waiting to happen to me.